### DISCUSSION TITLE
Dashboard - Cannot read properties of undefined (reading 'map')

### DISCUSSION THREAD

**Discussion by oneal32xx** [datalust/seq-tickets#1860] [Help] [answered]
Hi Team, i found some problem and need help 

### i need help 1.
in Seq Dashboard default page, I found some Error message in chart. the message is 
```  'Cannot read properties of undefined (reading 'map')' ``` 
```  'To plot a pie chart, the query must return two columns. The rightmost column must be numeric' ```
```  'To build a table, the query must return a non-aggregated rowset.' ```

 please check as shown below.

### i need help 2. 
when I was  selected another time range, the SQL ``` group by ``` section  not change anything, 

always show me   ``` $time ``` ,  where can i changed that QAQ!  please give me some tips.  thanks!

As shown below, when ever I select,  the ``` group by time ``` always show me  ``` $time ``` 
<hr>

 <b>Last 5 min</b>
<img width="1208" alt="截圖 2023-06-01 上午11 20 12" src="https://github.com/datalust/seq-tickets/assets/20452150/3e7c4f96-cf7e-4dc5-a2f6-09cf08f3b694">

 <b>Last 30 min</b>
<img width="1208" alt="截圖 2023-06-01 上午11 55 41" src="https://github.com/datalust/seq-tickets/assets/20452150/efe7503e-6302-4d1e-bbff-eeb7c185a985">

I using Seq version 2023.1.9101 
in windows server 2019

<img width="328" alt="截圖 2023-06-01 上午11 33 21" src="https://github.com/datalust/seq-tickets/assets/20452150/e4cb1da3-07b5-4e3d-86aa-250bba087dd4">

**Comment by nblumhardt** [comment]
Hi! Thanks for dropping us a line.

Something looks very broken, here. Could you please let us know which web browser and version you're using, as well as collect any errors shown in the browser developer tools console?

Many thanks
Nick

**Reply by oneal32xx** [reply]
Hi Nick , thanks for reply!!

Im used  thoes two device, same problem occurs
```Google Chrome version 113.0.5672.126. in MacOS Monterey 12.3.1 ``` 
```Google Chrome version 113.0.5672.127. in Windows 10 ```

<img width="1157" alt="截圖 2023-06-01 下午1 08 01" src="https://github.com/datalust/seq-tickets/assets/20452150/8fafc182-ab1a-4364-aae8-54c3dc3564ce">


Many thanks
Hans

**Reply by nblumhardt** [reply]
Thanks for the follow-up, that's interesting....

Could you please update to the latest Seq release, so that we can rule out any frontend issues we might already have fixed? The missing `map` function looks like a fairly standard type of frontend bug, but since this code path is very very well-used it's not obvious where the problem might be.

Thanks!

**Reply by oneal32xx** [reply]
thanks for your reply!
i will update to  v2023.1.9229  and try it again

**Reply by oneal32xx** [reply]
Hi Nick ,  
thanks for your help

i trying uninstall and  delete all files in seq folder,
and then reinstall  v2023.1.9229,  dashboard still broken....

I remembered that server's iis has created an application and used a reverse proxy to port 5341
So I tried creating a new app again with the same settings and the dashboard worked fine

Now I have two Seq platforms, one dashboard is working fine but the other dashboard is broken...

**Reply by nblumhardt** [reply, marked as answer]
Thanks for the follow-up! Is the conclusion that the problem is related to the reverse proxy, then?

**Reply by oneal32xx** [reply]
Yes, I think it is related!

**Comment by liammclennan** [comment]
Hi @oneal32xx 

Remove the `group by time($time)`. Dashboard time groupings are set by the dashboard controls and applied to all charts on the dashboard.

**Reply by oneal32xx** [reply]
Hi  @liammclennan
thanks for reply!!

dashboard query setting for ```group by```  are empty
where can I remove this 

<img width="1187" alt="截圖 2023-06-01 下午1 55 11" src="https://github.com/datalust/seq-tickets/assets/20452150/c90397ba-f69d-4419-a461-7b0d74f85d46">
